### ISSUE TITLE
error FS0192 : internal error : Length of Nullable metadata and needs of its processing do not match:  { Data = [|0uy; 1uy|]  Idx = 2 }

### ISSUE THREAD

**Issue by Evangelink** [dotnet/fsharp#17701]
When working on https://github.com/YoloDev/YoloDev.Expecto.TestSdk/pull/152 I suddenly started to get many nullability related failures.

I am hitting this TODO https://github.com/dotnet/fsharp/blob/c228ace3da3f1e90adeac913a30f7b42920d59e5/src/Compiler/Checking/import.fs#L269-L271

**Repro steps**

Provide the steps required to reproduce the problem:

1. Clone https://github.com/YoloDev/YoloDev.Expecto.TestSdk/pull/152
2. `dotnet build`

**Expected behavior**

No error

**Actual behavior**

error FS0192 : internal error : Length of Nullable metadata and needs of its processing do not match:  { Data = [|0uy; 1uy|]  Idx = 2 }

**Known workarounds**

Disable nullability

**Related information**

SDKS:
3.1.426 [C:\Program Files\dotnet\sdk]
6.0.425 [C:\Program Files\dotnet\sdk]
7.0.410 [C:\Program Files\dotnet\sdk]
8.0.100 [C:\Program Files\dotnet\sdk]
8.0.304 [C:\Program Files\dotnet\sdk]
9.0.100-preview.7.24407.12 [C:\Program Files\dotnet\sdk]
9.0.100-rc.1.24452.12 [C:\Program Files\dotnet\sdk]

**Comment by T-Gro** [comment]
Thanks for the report.
One of the linked C# dependencies used a nullness metadata format which the F# compiler cannot process, will be some nested generic type which was added via nuget and consumed.

I will check if I can find it and figure the missing case to be handled.

I noticed the linked library uses System.Nullable at a few places, hat would be my first guess (some nested generics involving System.Nullable)
